qiskit_ibm_runtime/base_primitive.py

```diff
@@ -154,16 +169,30 @@ def _run_primitive(self, primitive_inputs: Dict, user_kwargs: Dict) -> RuntimeJo
 
         combined = Options._set_default_resilience_options(combined)
         combined = Options._remove_none_values(combined)
-
         primitive_inputs.update(Options._get_program_inputs(combined))
 
-        if self._backend and combined["transpilation"]["skip_transpilation"]:
+        if self._backend and combined["transpilation"]["skip_transpilation"] and not run_simulator:
```

Comment:
`check_faulty` should apply to fake backends as well, if we ever get one that mimics a real backend with faulty edge/qubits

Reply:
There is currently no `check_faulty` method in `FakeBackendV2` nor in `AerSimulator`. To support this, we would need to add such a method to `FakeBackendV2` after it is transferred to Runtime. So I suggest we defer this to a separate issue.